lib/PuppeteerSharp/Request.cs

```diff
@@ -230,24 +230,31 @@ public async Task RespondAsync(ResponseData response)
 
             _interceptionHandled = true;
 
-            var responseHeaders = new Dictionary<string, string>();
+            var responseHeaders = new List<KeyValuePair<string, string>>();
```

Comment:
Did you try using a passing `StringComparer.OrdinalIgnoreCase` to the Dictionary?
I would like to avoid having another property.

Reply:
Hi @kblok by passing that we can make Dictionary case insensitive but here we want Duplicate keys
like Set cookies header name can occur twice 
any further advice?

Reply:
Yeah, that's true, good catch.